Cabal/Distribution/Simple/Utils.hs

```diff
@@ -1415,14 +1415,14 @@ dropExeExtension filepath =
     (filepath', extension) | extension `elem` exeExtensions -> filepath'
                            | otherwise                      -> filepath
 
--- | List of possible executable file extensions on the current platform.
+-- | List of possible executable file extensions on the current build platform.
 exeExtensions :: [String]
 exeExtensions = case buildOS of
   -- Possible improvement: on Windows, read the list of extensions from the
   -- PATHEXT environment variable. By default PATHEXT is ".com; .exe; .bat;
   -- .cmd".
-  Windows -> ["", "exe"]
-  Ghcjs   -> ["", "exe"]
+  Windows -> ["", ".exe"]
```

Comment:
and as I think it, it's confusing that we have **both**, this and https://hackage.haskell.org/package/Cabal-3.0.0.0/docs/src/Distribution.Simple.BuildPaths.html#exeExtension without any cross-reference.

It would be confusing if one has dot, and another doesn't.

Reply:
This whole extensions buissness is just inconsistent as hell, consider this:

```
splitExtension "foo." == ("foo", ".")
"." `isExtensionOf` "foo." == True -- So far so good

splitExtension "foo" == ("foo", "")
"" `isExtensionOf` "foo" == False -- Ugh...
```
If the dot in an extension wasn't optional things could be nice and consistent, where `""` stands for no extension and `"."` stands for the empty extension. So I'm not sure what to do here, pretty much all the options are gross and ad-hoc.